lock-impl/src/main/java/com/palantir/lock/logger/LockServiceStateLogger.java

```diff
@@ -146,17 +105,17 @@ private Map<String, Object> generateOutstandingLocksYaml(
                     requestList.forEach(request -> {
                         outstandingRequestMap.putIfAbsent(
                                 request.getLockDescriptor(),
-                                new SimpleLockRequestsWithSameDescriptor(request.getLockDescriptor()));
-                        outstandingRequestMap.get(request.getLockDescriptor()).addLockRequest(request);
+                                ImmutableSimpleLockRequestsWithSameDescriptor.builder()
+                                        .lockDescriptor(request.getLockDescriptor()));
+                        outstandingRequestMap.get(request.getLockDescriptor()).addLockRequests(request);
                     });
                 });
             }
         });
+        List<SimpleLockRequestsWithSameDescriptor> aggregatedRequests = new ArrayList<>();
+        outstandingRequestMap.forEach((_descriptor, builder) -> aggregatedRequests.add(builder.build()));
 
-        List<SimpleLockRequestsWithSameDescriptor> sortedOutstandingRequests =
-                sortOutstandingRequests(outstandingRequestMap.values());
-
-        return nameObjectForYamlConversion(OUTSTANDING_LOCK_REQUESTS_TITLE, sortedOutstandingRequests);
+        return sortOutstandingRequests(aggregatedRequests);
```

Comment:
nit: can we remove `sortOutstandingRequests` & the multiple collection copies and just stream through:

```suggestion
        return outstandingRequestMap.values().stream()
                .map(ImmutableSimpleLockRequestsWithSameDescriptor.Builder::build)
                .sorted(Comparator.comparing(SimpleLockRequestsWithSameDescriptor::getLockRequestsCount))
                .collect(Collectors.toList());
```